File: haskell/rna-transcription/DNA.hs
module DNA ( toRNA ) where 


rnafun :: Char -> Char
rnafun 'C' = 'C'
rnafun 'G' = 'G'
rnafun 'A' = 'A'
rnafun 'T' = 'U'

toRNA :: String -> String
toRNA  = map rnafun  

File: ocaml/rna-transcription/dna.ml
open Core.Std

type dna = [ `A | `C | `G | `T ]
type rna = [ `A | `C | `G | `U ]

let rnafun : dna -> rna = function
  | `C -> `G
  | `G -> `C
  | `A -> `U
  | `T -> `A

let to_rna  = List.map ~f:rnafun
